corehq/apps/userreports/reports/builder/forms.py

```diff
@@ -102,19 +102,25 @@ def render(self, name, value, attrs=None, choices=()):
         final_attrs = self.build_attrs(attrs, name=name)
 
         return format_html(
-            '<input{0} data-bind="'
-            '   questionsSelect: {1},'
-            '   value: \'{2}\','
-            '   optionsCaption: \' \''
-            '"/>',
+            """
+            <input{0} data-bind='
+               questionsSelect: {1},
+               value: "{2}",
+               optionsCaption: " "
+            '/>
+            """,
             flatatt(final_attrs),
             mark_safe(self.render_options(choices)),
             value
         )
 
     def render_options(self, choices):
+
+        def escape(literal):
+            return literal.replace('&', '&amp;').replace("'", "&#39;")
```

Comment:
Yeah, I'm guessing if you need to escape "&", then you probably need to escape other characters too. I think `cgi.escape` will escape double quotes but not single quotes. `django.utils.html.escape` will escape both kinds of quotes. That's probably what you want, or you could use `cgi.escape` and switch your outer quotes to double quotes, i.e.

    """
    <input{0} data-bind="
        questionsSelect: {1},
        value: '{2}',
        optionsCaption: ' '
    "/>
    """,